### PR TITLE
Add alternative standin for AMR-wind that runs FLORIS

### PR DESCRIPTION
**Pull request by misi9170**
This PR adds a new standin for AMR-Wind that runs steady-state FLORIS calculations for the current wind speed, wind direction, and yaw angles. 

Running this requires that users clone [FLORIS](https://github.com/NREL/floris) and check out the [v4 branch](https://github.com/NREL/floris/tree/v4). Once FLORIS v4 is officially released around March--April 2024, this code will run using FLORIS installed directly from PyPI. 

At that stage, I will either submit a small follow-up PR to add FLORIS v4 to setup.py and ensure full compatibility.

TODO:
- [x] add tests for new functionality
- [X] Add simple example for functionality (example 08)
- [x] Add more thorough example for functionality in [WHOC](https://github.com/NREL/wind-hybrid-open-controller) (opened PR, see [here](https://github.com/NREL/wind-hybrid-open-controller/pull/26))
- [x] Add handling for externally-supplied standin data

### CODE REVIEW

**Comment by misi9170** [comment]
Addresses #63

**Comment by misi9170** [comment]
@genevievestarke @paulf81 This is now ready for review

**Review by paulf81 — APPROVED** [review]
Thank you @misi9170 , I reviewed all the files and this looks good to me, thank you for pointing at floris v4

File: example_case_folders/08_floris_only/amr_input.inp [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+samplingLine.z_line.type = LineSampler
+samplingLine.z_line.num_points = 128
+samplingLine.z_line.start = 5.0 5.0 5.0
+samplingLine.z_line.end = 5.0 5.0 1275.0
+
+#¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨¨#
+#              TURBINES                 #
+#.......................................#
+
+Actuator.type = UniformCtDisk
```
Comment (genevievestarke) [inline example_case_folders/08_floris_only/amr_input.inp:149]: We have two definitions of turbines in here that is messing up the turbine definition in the amr_standin.  Should we only allow only one turbine definition, or adjust the way it's read?

File: hercules/amr_wind_standin.py [hunk 2]
```diff
@@ -184,12 +184,16 @@ def run(self):
...
 
             # Compute the turbine power using a simple formula
+            if self.message_from_server is not None:
+                yaw_angles = self.message_from_server[-self.num_turbines :]
+            else:
+                yaw_angles = None
             (
                 amr_wind_speed,
                 amr_wind_direction,
                 turbine_powers,
```
Comment (genevievestarke) [inline hercules/amr_wind_standin.py:194]: We need to decide what units power is in.  It seems to be mostly in kW, I think, but this seems to give power back in watts from Floris?

**Review by genevievestarke — CHANGES_REQUESTED** [review]
It looks good and it runs, we just need to discuss the turbine definitions and power units before it's merged!

File: example_case_folders/08_floris_only/amr_input.inp (hunk 1 above)
Comment (misi9170) [reply example_case_folders/08_floris_only/amr_input.inp:149]: I'll enforce that the Actuator type is UniformCtDisk for the FLORIS stand-in and remove the JoukowskyDisk information below.

Comment (misi9170) [reply example_case_folders/08_floris_only/amr_input.inp:149]: I'll remove the JoukowskyDisk definition here.

File: hercules/amr_wind_standin.py (hunk 2 above)
Comment (misi9170) [reply hercules/amr_wind_standin.py:194]: After discussion, moving forward with power being specified in kW across all modules. I'll update the FLORIS stand-in to reflect this.

Comment (misi9170) [reply hercules/amr_wind_standin.py:194]: @genevievestarke I've now updated the Floris stand-in code to convert directly to kW on receiving the powers from Floris (tests also updated to match)

File: example_case_folders/08_floris_only/amr_input.inp (hunk 1 above)
Comment (misi9170) [reply example_case_folders/08_floris_only/amr_input.inp:149]: @genevievestarke `read_amr_wind_input()` on `Emulator()` now checks for the specified `actuator_type` and gets the correct `rotor_diameter`. I checked this before removing the JoukowskyDisk definition for the input file and it ran as expected; then I went ahead and removed the JoukowskyDisk in this example.

**Review by genevievestarke — APPROVED** [review]
Looks good to me, I think we can merge in!

**Pull request merged**